code/datums/mind.dm

```diff
@@ -317,7 +317,7 @@
 		P = locate() in PDA
 	if (!P) // If we couldn't find a pen in the PDA, or we didn't even have a PDA, do it the old way
 		P = locate() in all_contents
-		if(!P) // I do not have a pen.
+		if(!P && (traitor_mob.client.prefs.uplink_spawn_loc == UPLINK_PEN)) // I have no pen and I must write.

```

Comment:
Move this whole section of code down by around 30 lines into the switch.
Specifically after line 345

Reply:
Done. Thanks!